pkg/ddc/juicefs/utils.go

```diff
@@ -305,3 +305,61 @@ func GetMetricsPort(options map[string]string) (int, error) {
 
 	return int(port), nil
 }
+
+func parseVersion(version string) (*ClientVersion, error) {
+	if version == common.NightlyTag {
+		return &ClientVersion{
+			Tag: common.NightlyTag,
+		}, nil
+	}
+	re := regexp.MustCompile(`^v?(\d+)\.(\d+)\.(\d+)(?:-(.+))?$`)
+	matches := re.FindStringSubmatch(strings.TrimSpace(version))
+	if matches == nil {
+		return nil, fmt.Errorf("invalid version string: %s", version)
+	}
+	major, err := strconv.Atoi(matches[1])
+	if err != nil {
+		return nil, fmt.Errorf("invalid major version: %s", matches[1])
+	}
+	minor, err := strconv.Atoi(matches[2])
+	if err != nil {
+		return nil, fmt.Errorf("invalid minor version: %s", matches[2])
+	}
+	patch, err := strconv.Atoi(matches[3])
+	if err != nil {
+		return nil, fmt.Errorf("invalid patch version: %s", matches[3])
+	}
+	return &ClientVersion{
+		Major: major,
+		Minor: minor,
+		Patch: patch,
+		Tag:   matches[4],
```

Comment:
I think the length should be checked to avoid index out of array error

Reply:
done